osu.Game/Screens/Select/Leaderboards/Leaderboard.cs

```diff
@@ -164,5 +276,85 @@ protected override void Update()
                 }
             }
         }
+
+        private class RetryButton : BeatSyncedContainer
+        {
+            private readonly SpriteIcon icon;
+
+            public Action Action;
+
+            private OsuColour colours;
+
+            public RetryButton()
+            {
+                Height = 26;
+                Width = 26;
+                Child = new OsuClickableContainer
+                {
+                    AutoSizeAxes = Axes.Both,
+                    Anchor = Anchor.Centre,
+                    Origin = Anchor.Centre,
+                    Action = () => Action?.Invoke(),
+                    Child = icon = new SpriteIcon
+                    {
+                        Icon = FontAwesome.fa_refresh,
+                        Size = new Vector2(26),
+                        Shadow = true,
+                    },
+                };
+            }
+
+            [BackgroundDependencyLoader]
+            private void load(OsuColour colours)
+            {
+                this.colours = colours;
+            }
+
+            protected override void OnNewBeat(int beatIndex, TimingControlPoint timingPoint, EffectControlPoint effectPoint, TrackAmplitudes amplitudes)
+            {
+                var duration = timingPoint.BeatLength / 2;
+
+                icon.Animate(
+                    i => i.MoveToY(-3, duration, Easing.Out),
+                    i => i.ScaleTo(IsHovered ? 1.3f : 1.1f, duration, Easing.Out)
+                ).Then(
+                    i => i.MoveToY(0, duration, Easing.In),
+                    i => i.ScaleTo(IsHovered ? 1.4f : 1f, duration, Easing.In)
+                );
+            }
+
+            protected override bool OnHover(InputState state)
+            {
+                icon.ScaleTo(1.4f, 400, Easing.OutQuint);
+                return base.OnHover(state);
+            }
+
+            protected override void OnHoverLost(InputState state)
+            {
+                icon.ClearTransforms();
+                icon.ScaleTo(1f, 400, Easing.OutQuint);
+                base.OnHoverLost(state);
+            }
+
+            protected override bool OnMouseDown(InputState state, MouseDownEventArgs args)
+            {
+                icon.FadeColour(colours.Yellow, 400);

```

Comment:
if anything i'd reverse the face and hover transforms in line with the rest of lazer..

in fact, if you make this whole class `: OsuHoverContainer` this will perform the on-hover fade for you.

then cahnge mousedown to scaleTo(0.8) and mouseup to scaleTo(1.0) or something similar to OsuButton